src/PrestaShopBundle/Bridge/Helper/AddFlashMessage.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Bridge\Helper;
+
+use Symfony\Component\HttpFoundation\Exception\SessionNotFoundException;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class AddFlashMessage
+{
+    private const TYPES = [
+        'errors' => 'error',
+        'warnings' => 'warning',
+        'informations' => 'info',
+        'confirmations' => 'success',
+    ];
+
+    public function __construct(private readonly RequestStack $requestStack)
+    {
+    }
+
+    public function __invoke(string $type, string $message): void
```

Comment:
Class with invoke method can be used as callback

Reply:
Hmm I tend to think like matthieu, the usage of callback doesn't seem to bring more feature nor help the readbility. Why do you prefer using this as a callback instead of doing something like:
```php
                $flashMessageBridge = SymfonyContainer::getInstance()->get(FlashMessageBridge::class);
                foreach ($this->$type as $message) {
                    $addFlashMessage->addMessage($type, $message);
                }
                $this->$type = [];
```

Reply:
I disagree but it's up to you guys